mpf/devices/multiball.py

```diff
@@ -147,6 +143,52 @@ def start(self):
             balls: The number of balls in this multiball
         '''
 
+    def timer_start(self) -> None:
+        """Start the timer.
+
+        This is started when multiball starts if configured.
+        """
+        self.machine.events.post('ball_save_{}_timer_start'.format(self.name))
+        '''event: ball_save_(name)_timer_start
+        desc: The multiball ball save called (name) has just start its countdown timer.
+        '''
+
+        shoot_again_ms = self.config['shoot_again'].evaluate([])
+        grace_period_ms = self.config['grace_period'].evaluate([])
+        hurry_up_time_ms = self.config['hurry_up_time'].evaluate([])
+        if shoot_again_ms > 0:
+            self.debug_log('Starting ball save timer: %ss',
+                           shoot_again_ms)
+            # Register stop handler
+            self.delay.add(name='disable_shoot_again',
+                           ms=(shoot_again_ms +
+                               grace_period_ms),
+                           callback=self.stop)
+            self.delay.add(name='grace_period',
```

Comment:
I guess those should also be removed in stop and post the event right away if they did not fire before, right? Otherwise, they will fire slightly too late which might lead to weird animations or similar.